### PR TITLE
Auth api

### PR DESCRIPTION
**Pull request by oscar-g**
This is incomplete, but I want to get some discussion started.

First, I made a couple html files that I use to test this with: https://drive.google.com/file/d/0B3eMhCTHyMkvSFdwa2FZa2N3LVU/view?usp=sharing

I use an apache virtual server to host those html files and CORS is set up in `helpers/auth.js`

The api is able to receive a Github oAuth `code` and retreive an `access_token` using that code. The access token is used to get the user's profile.

I'm working on hooking-up the successful retrieval of the access token to the user model and the client's session.
I also want to implement a json web token scheme.

### CODE REVIEW

**Comment by evanplaice** [comment]
@oscar-g Why merge in develop instead of rebase?

**Comment by oscar-g** [comment]
@evanplaice: I'm not sure that helmet let's you do that, but I'll look into it...check out this thread: https://github.com/helmetjs/helmet/issues/91

Also, I did merge because it gives a better view of the history and it is non-destructive...I guess it just makes the commit log a bit cluttered.  I'll use rebase from now on, except when we'll benefit from seeing the merge history.

**Comment by evanplaice** [comment]
@oscar-g Looks like you were right about CORS https://github.com/helmetjs/helmet/issues/79

:+1: on using rebase. With multiple feature branches under active development it makes for a much cleaner history.

**Comment by thebergamo** [comment]
For me its fine. =D
I don't see anything correlated in Helmet, corser is pretty nice I think.

File: auth/index.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+function isActiveStrategy (req, res, next) {
+  let active = false;
+
+  active_strategies.forEach((strategy) => {
+    if (req.params.strategy.toString() === strategy.endpoint) {
+      active = true;
+    }
+  });
+
+  if (active === true) {
```
Comment (007lva) [inline auth/index.js]: SUG:
```if (active) {```

**Pull request merged**